components/ca-select.js

```diff
@@ -0,0 +1,367 @@
+         * @returns {void} void
+         */
+        render() {
+
+            if (!this.root) return;
+
+            const self = this;
+            let data = this.data || [];
+            const hasData = (Array.isArray(data) && this.data.length);
+            const defaultLabel = (hasData) ? ((this.type == 'tel') ? 'Code' : 'Please select') : 'No options available';
+
+            if (this.readonly && this.style !== 'checkbox') {
+
+                this.root.setAttribute('readonly', true);
+            }
+
+            // render options if we have some data to work with
+            if (hasData) {
+
+                if (this.style !== 'checkbox') {
+
+                    // add default (unselectable) option
+                    createElement(this.root, 'option', { value: '', 'data-index': 0, selected: true }, defaultLabel);
+                }
+
+                if (this.style == 'checkbox') {
+                    
+                    this.rootCheckbox = createElement(self.root, 'div', {});
+                }
+
+                for (let i = 0, l = this.data.length; i < l; i++) {
+
+                    const item = this.data[i];
+
+                    switch (this.style) {
+                        
+                        case 'checkbox': {
+
+                            const p = createElement(this.rootCheckbox, 'p', {});
+                            const opt = createElement(p, 'input', { type: 'checkbox', value: item.value, 'data-index': i+1 });
+                            p.appendChild(document.createTextNode(' ' + item.label + ' '));
+
+                            if (item.attachment) {
+
+                                createElement(p, 'a', { href: item.attachment.href, target: '_blank' }, item.attachment.title);
+                            }
+
+                            opt.addEventListener('change', () => {
+
+                                this.setAttribute('data-selected', (this.checked).toString());
+                                onChangeHandler.call(self);
+                            });
+                        } break;
+
+                        default: {
+                            const opt = createElement(this.root, 'option', { value: item.value, 'data-index': i+1 }, item.label);
+
+                            if (this.multiple) {
+
+                                this.root.querySelectorAll('option')[0].disabled = true;
+                            }
+
+                            opt.selected = (item.value === this.value);
+                                
+                            if (!this.isTouch && this.multiple) {
+
+                                opt.onmousedown = e => {
+
+                                    helpers.cancelEvent(e || event);
+                                    self.checkChange.call(self, this);
+                                }
+
+                                this.root.onclick = e => {
+
+                                    helpers.cancelEvent(e || event);
```

Comment:
helpers.cancelEvent should be replaced by adding the cancel-event module as a dependency